jvb/src/main/kotlin/org/jitsi/videobridge/Endpoint.kt

```diff
@@ -1172,6 +1209,80 @@ class Endpoint @JvmOverloads constructor(
         private val random = SecureRandom()
     }
 
+    private inner class SctpCallbacks(transport: DcSctpTransport) : DcSctpBaseCallbacks(transport) {
+        override fun sendPacketWithStatus(packet: ByteArray): SendPacketStatus {
+            val newBuf = ByteBufferPool.getBuffer(packet.size)
+            System.arraycopy(packet, 0, newBuf, 0, packet.size)
+
+            sctpSendPcap.observe(newBuf, 0, packet.size)
+            dtlsTransport.sendDtlsData(newBuf, 0, packet.size)
+
+            return SendPacketStatus.kSuccess
+        }
+
+        override fun OnMessageReceived(message: DcSctpMessage) {
+            // We assume all data coming over SCTP will be datachannel data
+            val dataChannelPacket = DataChannelPacket(message)
+            // Post the rest of the task here because the current context is
+            // holding a lock inside the SctpSocket which can cause a deadlock
+            // if two endpoints are trying to send datachannel messages to one
+            // another (with stats broadcasting it can happen often)
+            incomingDataChannelMessagesQueue.add(PacketInfo(dataChannelPacket))
+        }
+
+        override fun OnError(error: ErrorKind, message: String) {
+            logger.warn("SCTP error $error: $message")
+        }
+
+        override fun OnAborted(error: ErrorKind, message: String) {
+            logger.warn("SCTP aborted with error $error: $message")
+        }
+
+        override fun OnConnected() {
+            logger.info("SCTP connection is ready, creating the Data channel stack")
+            val dataChannelStack = DataChannelStack(
+                { data, sid, ppid ->
+                    val message = DcSctpMessage(sid.toShort(), ppid, data.array())
+                    val status = sctpTransport?.socket?.send(message, DcSctpTransport.DEFAULT_SEND_OPTIONS)
+                    return@DataChannelStack if (status == SendStatus.kSuccess) {
+                        0
+                    } else {
+                        logger.error("Error sending to SCTP: $status")
+                        -1
+                    }
+                },
+                logger
+            )
+            this@Endpoint.dataChannelStack = dataChannelStack
+            // This handles if the remote side will be opening the data channel
+            dataChannelStack.onDataChannelStackEvents { dataChannel ->
+                logger.info("Remote side opened a data channel.")
+                messageTransport.setDataChannel(dataChannel)
+            }
+            dataChannelHandler.setDataChannelStack(dataChannelStack)
+            if (OPEN_DATA_CHANNEL_LOCALLY) {
```

Comment:
OPEN_DATA_CHANNEL_LOCALLY is just hard-coded to `false`. If this mode hasn't been tested should we just remove it?

Reply:
Probably.  We have real code to open the datachannel locally in Relay, so we don't need this as a model for the future.